x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/grid_embeddable/grid_embeddable.tsx

```diff
@@ -0,0 +1,758 @@
+
+    const actionColumn: EuiTableActionsColumnType<FieldVisConfig> = {
+      name: (
+        <FormattedMessage
+          id="xpack.dataVisualizer.index.dataGrid.actionsColumnLabel"
+          defaultMessage="Actions"
+        />
+      ),
+      actions,
+      width: '70px',
+    };
+
+    return [actionColumn];
+  }, [input.indexPattern, services, searchQueryLanguage, searchString]);
+
+  return {
+    configs,
+    searchQueryLanguage,
+    searchString,
+    searchQuery,
+    extendedColumns,
+    documentCountStats,
+    metricsStats,
+  };
+};
+
+export const DiscoverWrapper = ({
+  input,
+  onOutputChange,
+}: {
+  input: DataVisualizerGridEmbeddableInput;
+  onOutputChange?: (ouput: any) => void;
+}) => {
+  const [dataVisualizerListState, setDataVisualizerListState] =
+    useState<Required<DataVisualizerIndexBasedAppState>>(restorableDefaults);
+
+  const onTableChange = useCallback(
+    (update: DataVisualizerTableState) => {
+      setDataVisualizerListState({ ...dataVisualizerListState, ...update });
+      if (onOutputChange) {
+        onOutputChange(update);
+      }
+    },
+    [dataVisualizerListState, onOutputChange]
+  );
+  const { configs, searchQueryLanguage, searchString, extendedColumns } = useDataVisualizerGridData(
+    input,
+    dataVisualizerListState
+  );
+  const getItemIdToExpandedRowMap = useCallback(
+    function (itemIds: string[], items: FieldVisConfig[]): ItemIdToExpandedRowMap {
+      return itemIds.reduce((m: ItemIdToExpandedRowMap, fieldName: string) => {
+        const item = items.find((fieldVisConfig) => fieldVisConfig.fieldName === fieldName);
+        if (item !== undefined) {
+          m[fieldName] = (
+            <IndexBasedDataVisualizerExpandedRow
+              item={item}
+              indexPattern={input.indexPattern}
+              combinedQuery={{ searchQueryLanguage, searchString }}
+            />
+          );
+        }
+        return m;
+      }, {} as ItemIdToExpandedRowMap);
+    },
+    [input, searchQueryLanguage, searchString]
+  );
+
+  return (
+    <DataVisualizerTable<FieldVisConfig>
+      items={configs}
+      pageState={dataVisualizerListState}
+      updatePageState={onTableChange}
+      getItemIdToExpandedRowMap={getItemIdToExpandedRowMap}
+      extendedColumns={extendedColumns}
+      showPreviewByDefault={input?.showPreviewByDefault}
+      onChange={onOutputChange}
+    />
+  );
+
+  return <div />;
+};
+
+export const IndexDataVisualizerViewWrapper = (props: {
+  id: string;
+  embeddableContext: InstanceType<IDataVisualizerGridEmbeddable>;
+  embeddableInput: Readonly<Observable<DataVisualizerGridEmbeddableInput>>;
+  onOutputChange?: (output: any) => void;
+}) => {
+  const { embeddableInput, onOutputChange } = props;
+
+  const input = useObservable(embeddableInput);
+  if (input && input.indexPattern) {
+    return <DiscoverWrapper input={input} onOutputChange={onOutputChange} />;
+  } else {
+    return <div />;
```

Comment:
so if there's no index pattern there's just a `div` rendered, right? think it might make sense to render an error message in this case even if this case is unlikely

Reply:
Updated here https://github.com/elastic/kibana/pull/107184/commits/ba1cad7ff6ee7b3391fb42000306c6b3b96096d5